apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/boot/ServiceManager.java

```diff
@@ -36,11 +38,13 @@
     INSTANCE;
 
     private static final ILog LOGGER = LogManager.getLogger(ServiceManager.class);
+    public static Instrumentation INSTRUMENTATION;
```

Comment:
Why this?

Reply:
In order to make it more convenient to use instrumentation where necessary.

Reply:
This is not recommended way to expose. Metadata like this should be loaded by core directly. Not a service should do.